File: tooling/archetypes/camel-archetype-component/src/main/resources/archetype-resources/src/test/java/__name__ComponentTest.java
package ${package};

import org.apache.camel.builder.RouteBuilder;
import org.apache.camel.component.mock.MockEndpoint;
import org.apache.camel.test.junit4.CamelTestSupport;
import org.junit.Test;

public class ${name}ComponentTest extends CamelTestSupport {

    @Test
    public void test${name}() throws Exception {
        MockEndpoint mock = getMockEndpoint("mock:result");
        mock.expectedMinimumMessageCount(1);       
        
        assertMockEndpointsSatisfied();
    }

    @Override
    protected RouteBuilder createRouteBuilder() throws Exception {
        return new RouteBuilder() {
            public void configure() {
                from("${scheme}://foo")
                  .to("${scheme}://bar")
                  .to("mock:result");
            }
        };
    }
}

File: tooling/archetypes/camel-archetype-component-scala/src/main/resources/archetype-resources/src/test/scala/__name__ComponentTest.scala
package ${package}

import org.apache.camel.component.mock.MockEndpoint
import org.apache.camel.test.junit4.CamelTestSupport
import org.junit.Test
import org.apache.camel.scala.dsl.builder.{RouteBuilder, RouteBuilderSupport}

// The trait RouteBuilderSupport helps bridge the Scala RouteBuilder with the Java RouteBuilder,
// that the CamelTestSupport expects and uses
class ${name}ComponentTest extends CamelTestSupport with RouteBuilderSupport {

  // Create a simple route that uses both the consumer and producer endpoint of the new component.
  override def createRouteBuilder() = {
    new RouteBuilder() {

      "${scheme}://foo" to "${scheme}://bar" to "mock:result"
    }

  }

  @Test
  def test${name}() = {
    val mock: MockEndpoint = getMockEndpoint("mock:result")
    mock.expectedMinimumMessageCount(1)

    assertMockEndpointsSatisfied()
  }

}
